app/src/main/java/org/gdg/frisbee/android/activity/MainActivity.java

```diff
@@ -203,31 +190,39 @@ protected void onResume() {
         }
     }
 
-    private void selectChapter(Chapter chapter) {
-        mViewPagerAdapter.setSelectedChapter(chapter);
-        getSupportActionBar().setSelectedNavigationItem(mSpinnerAdapter.getPosition(chapter));
-    }
-
-    private void initChapters(ArrayList<Chapter> chapters) {
+    /**
+     * Initializes ViewPager, SlidingTabLayout, and Spinner Navigation. 
+     * 
+     * If the selectedChapter is null, tries to get it from Intent Extras,
+     *                                  if not found, gets the first Chapter.
+     * This function is called after cache query or network call is success,
+     * or after an orientation change using savedInstance.
+     *  
+     * @param chapters Chapter array to be initialized, never null.
+     * @param selectedChapter selectedChapter, can be null.
+     */
+    private void initChapters(@NonNull ArrayList<Chapter> chapters,
+                              @Nullable Chapter selectedChapter) {
         addChapters(chapters);
-        Chapter chapter = null;
 
-        if (getIntent().hasExtra(Const.EXTRA_GROUP_ID)) {
-            String chapterId = getIntent().getStringExtra(Const.EXTRA_GROUP_ID);
+        if (selectedChapter == null && getIntent().hasExtra(Const.EXTRA_CHAPTER_ID)) {
+            final String chapterId = getIntent().getStringExtra(Const.EXTRA_CHAPTER_ID);
             for (Chapter c : chapters) {
                 if (c.getGplusId().equals(chapterId)) {
-                    chapter = c;
+                    selectedChapter = c;
                     break;
                 }
             }
-            if (chapter == null) {
-                chapter = chapters.get(0);
-            }
-        } else {
-            chapter = chapters.get(0);
         }
+        
+        if (selectedChapter == null) {
+            selectedChapter = chapters.get(0);
+        }
+
+        getSupportActionBar().setSelectedNavigationItem(mSpinnerAdapter.getPosition(selectedChapter));

```

Comment:
Nooooo!!! That doesn't exist anymore! I'll have to rebase that! :crying_cat_face: 
That method is deprecated and I've changed that since we won't be using an action bar anymore.
But it's okay, I'll deal with it.